jscomp/ml/error_message_utils.ml

```diff
@@ -204,7 +204,8 @@ let typeClashContextMaybeOption ty_expected ty_res =
   | ( {Types.desc = Tconstr (expectedPath, _, _)},
       {Types.desc = Tconstr (typePath, _, _)} )
     when Path.same Predef.path_option typePath
-         && Path.same expectedPath Predef.path_option = false ->
+         && Path.same expectedPath Predef.path_option = false 
+         && Path.same expectedPath Predef.path_uncurried = false ->
```

Comment:
Here was another gremlin I came across where you get a suggestion to unwrap your option. For the example in the tests/descriptioon you get:

```
  This value might need to be wrapped in a function that takes an extra
  parameter of type unit
                                                                                                                                            
  Here's the original error message
  This has type: option<'a>
  But it's expected to have type: (. unit) => option<int>
                                                                                                                                            
  Possible solutions:
  - Unwrap the option to its underlying value using `yourValue->Belt.Option.getWithDefault(someDefaultValue)`
```

Where the possible solution is not meant to be for functions but uncurried functions still pattern match on this case.